### PR TITLE
ENH: Add use_nullable_dtypes to read_clipboard

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/clipboards.py [hunk 1]
```diff
@@ -24,6 +26,19 @@ def read_clipboard(sep: str = r"\s+", **kwargs):  # pragma: no cover
...
+        implementation, even if no nulls are present.
+
+        The nullable dtype implementation can be configured by calling
+        ``pd.set_option("mode.dtype_backend", "pandas")`` to use
+        numpy-backed nullable dtypes or
+        ``pd.set_option("mode.dtype_backend", "pyarrow")`` to use
+        pyarrow-backed nullable dtypes (using ``pd.ArrowDtype``).
+
+        .. versionadded:: 2.0
+
```
Comment (gfyoung) [inline pandas/io/clipboards.py:41]: Don't we need to add that this is only supported when `engine="python"`?

File: pandas/tests/io/test_clipboard.py [hunk 2]
```diff
@@ -402,3 +409,62 @@ def test_raw_roundtrip(self, data):
...
+                "d": string_array_na,
+                "e": Series([2, NA], dtype="Int64"),
+                "f": Series([4.0, NA], dtype="Float64"),
+                "g": Series([NA, NA], dtype="Int64"),
+                "h": Series([True, False], dtype="boolean"),
+                "i": Series([False, NA], dtype="boolean"),
+            }
+        )
+        if dtype_backend == "pyarrow":
+            import pyarrow as pa
```
Comment (gfyoung) [inline pandas/tests/io/test_clipboard.py:458]: Why do we need this import again?

File: pandas/io/clipboards.py (hunk 1 above)
Comment (phofl) [reply pandas/io/clipboards.py:41]: cc @mroeschke Copied this from read_csv. IIRC you added the description there? Thoughts?

File: pandas/tests/io/test_clipboard.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/io/test_clipboard.py]: Copy paste... Thx, removed

File: pandas/io/clipboards.py (hunk 1 above)
Comment (mroeschke) [reply pandas/io/clipboards.py:41]: Yeah probably good to note what engines  `pd.set_option("mode.dtype_backend", "pyarrow")` is compatible with here (and read_csv too)

Comment (phofl) [reply pandas/io/clipboards.py:43]: Thx, added a comment for read_csv as well

**Comment by gfyoung** [comment]
LGTM pending CI

**Comment by phofl** [comment]
cc @mroeschke Would you mind having a quick look?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Comment by phofl** [comment]
Thanks

**Pull request merged**